### PR TITLE
PolyhedralGeometry: Allow solve_mixed to return an iterator

### PR DESCRIPTION
**Pull request by lkastner**

### CODE REVIEW

**Comment by thofma** [comment]
Love it! A test would be great.

**Comment by lkastner** [comment]
> Love it! A test would be great.

Unfortunately I cannot test the exact result, as the order may change. I could do some `sort` as it is done in the doctests or is it enough as is? I do compare the `nrows` or `length` though to the correct result.

File: src/PolyhedralGeometry/solving_integrally.jl [hunk 1]
```diff
@@ -1,11 +1,33 @@
...
 
+function solve_mixed(as::Type{SubObjectIterator{PointVector{fmpz}}}, A::fmpz_mat, b::fmpz_mat, C::fmpz_mat, d::fmpz_mat)
+    ncols(A) == ncols(C) || throw(ArgumentError("solve_mixed(A,b,C,d): A and C must have the same number of columns."))
+    nrows(A) == nrows(b) || throw(ArgumentError("solve_mixed(A,b,C,d): A and b must have the same number of rows."))
+    nrows(C) == nrows(d) || throw(ArgumentError("solve_mixed(A,b,C,d): C and d must have the same number of rows."))
+    ncols(b) == 1 || throw(ArgumentError("solve_mixed(A,b,C,d): b must be a matrix with a single column."))
+    ncols(d) == 1 || throw(ArgumentError("solve_mixed(A,b,C,d): d must be a matrix with a single column."))
+    eq = (Matrix{BigInt}(A), vec(Matrix{BigInt}(b)))
+    ineq = (Matrix{BigInt}(-C), vec(Matrix{BigInt}(-d)))
+    P = Polyhedron(ineq, eq)
```
Comment (benlorenz) [inline src/PolyhedralGeometry/solving_integrally.jl]:
```suggestion
    P = Polyhedron((-C, vec(-d)), (A, vec(b)))
```
This should avoid a few unnecessary conversion.

**Review by benlorenz — APPROVED** [review]

File: src/PolyhedralGeometry/solving_integrally.jl (hunk 1 above)
Comment (lkastner) [reply src/PolyhedralGeometry/solving_integrally.jl]: Thanks.

**Pull request merged**